library/aws/shared.ts

```diff
@@ -408,6 +411,16 @@ export function createNetworkResources(
     gateway_id: internet_gateway.id,
   });
 
+  // Don't go through GW for predefined services.
+  for (const service of network_config.vpc_endpoint_services || []) {
+    AR.createVpcEndpoint(tfgen, service, {
```

Comment:
Could you factor this inner-block as a free-standing helper function avoid adding it to the `NetworkConfig` type?

Reply:
I definitly can.